server/server/controllers/adminController.js

```diff
@@ -1 +1,114 @@
-// 
\ No newline at end of file
+const RestaurantModel = require("../models/restaurant");
+const DishModel = require("../models/dish");
+const OrderModel = require("../models/order")
+
+//get all dishes
+const getDishItems = async (req, res) => {
+  try {
+    const dishes = await DishModel.find();
+    res.json(dishes);
+  } catch (error) {
+    res.status(500).json({ message: error.message });
+  }
+};
+
+// Add new Restaurant
+const addRestaurant = async (req, res) => {
+  try {
+    const { name, restaurantLocation, imageURL, description, dishes } =
+      req.body;
+
+    const newRestaurant = new RestaurantModel({
+      name,
+      restaurantLocation,
+      imageURL,
+      description,
+      dishes: [...dishes],
+    });
+
+    await newRestaurant.save();
+
+    res
+      .status(201)
+      .json({ message: "Restaurant added successfully", newRestaurant });
+      console.error("Error adding restaurant:", error);
```

Comment:
Transfer This line of code under catch error (line 36)

Reply:
ok